### PR TITLE
Adds Vector3.TransformCoord #4587

### PR DESCRIPTION
**Pull request by rpsfonseca**
By reading the documentation of XMVector3TransformCoord and comparing the other Transform methods in MonoGame, I think this is what is asked on #4587 .

I assume this is the way to keep the w component 1.0.

### CODE REVIEW

**Comment by rpsfonseca** [comment]
This has been updated after discussion on the open issue.

File: MonoGame.Framework/MathHelper.cs [hunk 1]
```diff
@@ -240,8 +240,20 @@ public static float Min(float value1, float value2)
...
+        /// <summary>
+        /// Maps a coordinate back to w = 1.
+        /// </summary>
+        /// <param name="value1">Coordinate value</param>
+        /// <param name="value2">W value</param>
+        /// <returns>The coordinate value divided by w</returns>
+        public static float PerspectiveDivision(float value1, float value2)
+        {
+            var invW = 1f / value2;
+            return invW * value1;
```
[the remark this answers is not in the document]
Comment (rpsfonseca) [reply MonoGame.Framework/MathHelper.cs]: Got it! Will change it.

**Comment by rpsfonseca** [comment]
I updated the Vector3 class with the missing TransformCoord methods, as suggested.

**Comment by rpsfonseca** [comment]
@nkast Can't understand why the builds are failing... I didn't touch any MacOS code, and apparently there are few less tests than what there's supposed to be. Should I create tests for the TransformCoord methods?

**Comment by Jjagg** [comment]
@rpsfonseca This is a really old PR. You'll need to rebase on top of develop to get it building.

- First create a new branch from your develop: `git branch vec3-coord`
- Add MG upstream repo (if you haven't yet): `git remote add upstream https://github.com/MonoGame/MonoGame.git`
- Fetch changes from MG repo: `git fetch upstream`
- Reset your develop branch to MG develop: `git reset --hard upstream/develop`
- I'd recommend setting up a new pull request from the vec3-coord branch, because that way you can keep develop up to date without the hassle:
  - So then rebase your changes:
    ```
    git checkout vec3-coord
    git rebase develop
    ```
  - There might be some conflicts that you can manually resolve. (Since the changes are all in the same place it shouldn't be too bad. You could also just reapply the changes manually to the MG develop branch.)
  - Set up a new PR from the vec3-coord branch and close this one.

- Alternatively keep your changes on your develop branch and push to your remote:
  - `git merge vec3-coord`

**Comment by Jjagg** [comment]
@rpsfonseca I'm closing this. I think the easiest route is to apply the changes to a new branch. if you'd still like to add this feature.

**Comment by rpsfonseca** [comment]
@Jjagg I'd still like to add the feature.
So could I just save these changes onto a auxiliary file locally, discard the changes on the branch, fetch the develop branch to get it to the latest version and then add there the changes I stored in the auxiliary file?

**Comment by Jjagg** [comment]
You can copy-paste the added ljnes from the diff view right here.

**Pull request closed without merging**